hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestOnlineSlowLogProvider.java

```diff
@@ -0,0 +1,307 @@
+    Assert.assertEquals(slowLogPayloads.size(), 14);
+
+    // confirm strict order of slow log payloads
+    confirmPayloadParams(0, 3991, slowLogPayloads);
+    confirmPayloadParams(1, 3992, slowLogPayloads);
+    confirmPayloadParams(2, 3993, slowLogPayloads);
+    confirmPayloadParams(3, 3994, slowLogPayloads);
+    confirmPayloadParams(4, 3995, slowLogPayloads);
+    confirmPayloadParams(5, 3996, slowLogPayloads);
+    confirmPayloadParams(6, 3997, slowLogPayloads);
+    confirmPayloadParams(7, 3998, slowLogPayloads);
+    confirmPayloadParams(8, 3999, slowLogPayloads);
+    confirmPayloadParams(9, 4000, slowLogPayloads);
+    confirmPayloadParams(10, 3987, slowLogPayloads);
+    confirmPayloadParams(11, 3988, slowLogPayloads);
+    confirmPayloadParams(12, 3989, slowLogPayloads);
+    confirmPayloadParams(13, 3990, slowLogPayloads);
+
+    boolean isRingBufferCleaned = onlineSlowLogProvider.clearSlowLogPayloads();
+    Assert.assertTrue(isRingBufferCleaned);
+    LOG.debug("cleared the ringbuffer of Online Slow Log records");
+    slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    // confirm ringbuffer is empty
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithDefaultDisableConfig() throws Exception {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.unset(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY);
+    onlineSlowLogProvider = new OnlineSlowLogProvider(conf);
+    Assert.assertEquals(onlineSlowLogProvider.getSlowLogPayloads().size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    for (int i = 0; i < 300; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(2, TimeUnit.SECONDS);
+
+    List<SlowLogPayload> slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithDisableConfig() throws Exception {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, false);
+    onlineSlowLogProvider = new OnlineSlowLogProvider(conf);
+    Assert.assertEquals(onlineSlowLogProvider.getSlowLogPayloads().size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    for (int i = 0; i < 300; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+
+    List<SlowLogPayload> slowLogPayloads = onlineSlowLogProvider.getSlowLogPayloads();
+    Assert.assertEquals(slowLogPayloads.size(), 0);
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, true);
+  }
+
+  @Test
+  public void testOnlineSlowLogWithDefaultConfig() throws Exception {
```

Comment:
Is this test method necessary? It'll have to be re-written if we change the default size later on. It's logic is redundant to your earlier test over a smaller buffer size, isn't it?